File: kimchanho/2025/week4/BOJ_27964.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.stream.Collectors;

/**
 * @title 콰트로치즈피자 (27964)
 * @description 각기 다른 치즈토핑이 4개 이상이면 yummy, 아니면 sad
 * @author Chanho Kim
 * @see https://www.acmicpc.net/problem/27964
 */
public class BOJ_27964 {

  public static void main(String[] args) throws Exception {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    br.readLine(); // skip n
    int cheeseToppings = Arrays.stream(br.readLine().split(" "))
        .filter(s -> s.endsWith("Cheese"))
        .collect(Collectors.toSet())
        .size();
    System.out.println(cheeseToppings >= 4 ? "yummy" : "sad");
  }
}

File: kimchanho/2025/week4/BOJ_27964.ts
/**
 * @title 콰트로치즈피자 (27964)
 * @description 각기 다른 치즈토핑이 4개 이상이면 yummy, 아니면 sad
 * @author Chanho Kim
 * @see https://www.acmicpc.net/problem/27964
 */
const solution = (input: string[]) => {
  const toppings= input[1].split(" ").filter((s) => s.endsWith("Cheese"));
  const toppingSet: Set<string> = new Set(toppings);
  console.log(toppingSet.size >= 4 ? "yummy" : "sad");
};

// setting for baekjoon online judge
const input: string[] = [];
require("readline")
.createInterface({ input: process.stdin, output: process.stdout })
.on("line", (line: string) => input.push(line))
.on("close", () => solution(input));
